To: vim_dev@googlegroups.com
Subject: Patch 7.3.413
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.413
Problem:    Build warnings on MS-Windows.
Solution:   Add type casts. (Mike Williams)
Files:	    src/ex_getln.c, src/message.c, src/term.c


*** ../vim-7.3.412/src/ex_getln.c	2012-01-20 20:44:38.000000000 +0100
--- src/ex_getln.c	2012-01-26 12:58:54.000000000 +0100
***************
*** 5923,5929 ****
  							      hist[i].hisnum);
  		    if (vim_strsize(hist[i].hisstr) > (int)Columns - 10)
  			trunc_string(hist[i].hisstr, IObuff + STRLEN(IObuff),
! 				  (int)Columns - 10, IOSIZE - STRLEN(IObuff));
  		    else
  			STRCAT(IObuff, hist[i].hisstr);
  		    msg_outtrans(IObuff);
--- 5923,5929 ----
  							      hist[i].hisnum);
  		    if (vim_strsize(hist[i].hisstr) > (int)Columns - 10)
  			trunc_string(hist[i].hisstr, IObuff + STRLEN(IObuff),
! 			     (int)Columns - 10, IOSIZE - (int)STRLEN(IObuff));
  		    else
  			STRCAT(IObuff, hist[i].hisstr);
  		    msg_outtrans(IObuff);
*** ../vim-7.3.412/src/message.c	2012-01-23 20:48:34.000000000 +0100
--- src/message.c	2012-01-26 12:58:54.000000000 +0100
***************
*** 325,331 ****
      if (e + 3 < buflen)
      {
  	mch_memmove(buf + e, "...", (size_t)3);
! 	len = STRLEN(s + i) + 1;
  	if (len >= buflen - e - 3)
  	    len = buflen - e - 3 - 1;
  	mch_memmove(buf + e + 3, s + i, len);
--- 325,331 ----
      if (e + 3 < buflen)
      {
  	mch_memmove(buf + e, "...", (size_t)3);
! 	len = (int)STRLEN(s + i) + 1;
  	if (len >= buflen - e - 3)
  	    len = buflen - e - 3 - 1;
  	mch_memmove(buf + e + 3, s + i, len);
*** ../vim-7.3.412/src/term.c	2012-01-20 17:15:47.000000000 +0100
--- src/term.c	2012-01-26 12:58:54.000000000 +0100
***************
*** 5252,5258 ****
      char_u	*src;
  {
      int		i;
!     int		slen = STRLEN(src);
  
      for (i = 0; i < tc_len; ++i)
      {
--- 5252,5258 ----
      char_u	*src;
  {
      int		i;
!     int		slen = (int)STRLEN(src);
  
      for (i = 0; i < tc_len; ++i)
      {
*** ../vim-7.3.412/src/version.c	2012-01-26 11:45:25.000000000 +0100
--- src/version.c	2012-01-26 12:59:14.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     413,
  /**/

-- 
It doesn't really matter what you are able to do if you don't do it.
				(Bram Moolenaar)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
